### PR TITLE
Past Events with fully visible titles added

### PR DESCRIPTION
**Pull request by M4RC1N76**
- [Description](#description)
  - [Type of change](#type-of-change)
- [Checklist](#checklist)

out

# Description

This PR introduces Past Events list in the Event section of the App. 
-  It allows user to check **Past** events option.
- Displays list of **Past** events (without the description as it is in the **Upcoming** events list).
- The titles of the **Past** and **Upcoming** events are displayed in full.
- The App passed continuous integration tests.

Fixes SVA-86-Past-Events

## Type of change

Please delete options that are not relevant

- [x] New Feature (non-breaking change which adds functionality)

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings

### CODE REVIEW

**Review by DougieWougie — CHANGES_REQUESTED** [review]
You should add .vscode to your .gitignore then remove .vscode/launch.json from this commit - these are specific to your Visual Studio Code and aren't needed in the repository.

File: app/src/Containers/EventsContainer.tsx
Comment (joefhall) [inline app/src/Containers/EventsContainer.tsx:153]: I don't think the logic on this line (and its closing bracket on line 159) are needed.  (Did you plan to include an `else` for past events which you've removed?)

**Review by joefhall — CHANGES_REQUESTED** [review]
Nice one @M4RC1N76 and well done for sticking with it!

Functionality's working well, I've just added some comments on places where the code needs tidying or simplifying.

Did you decide not to include the stuff we did the other day on hiding past event dates and times, or is that for a different ticket?

**Comment by M4RC1N76** [comment]
I have done all the tasks from the review. I have re-requested the review. I was not sure if I have to ask everyone again so the re-request for review was sent only to Dougie and Joe.
Regards Marcin Borowski 

Sent from Yahoo Mail on Android 
 
  On Tue, 4 Oct 2022 at 10:33, Marcin ***@***.***> wrote:    Hi Joe, To be honest I have focused on reviewing your part of the code so I left Past Events aside (their are still in review section). I need to tidy up the parts of the App when I have left my comments (reminders to do something or investigate) as you suggested. Dougie also added a few suggestions for changes. Thanks for the specifics regarding the code and comments. I am on it right now.
Regards Marcin Borowski
    On Tuesday, 4 October 2022 at 10:23:52 BST, Joe Derry Hall ***@***.***> wrote:  
 
 


@joefhall requested changes on this pull request.

Nice one @M4RC1N76 and well done for sticking with it!

Functionality's working well, I've just added some comments on places where the code needs tidying or simplifying.

Did you decide not to include the stuff we did the other day on hiding past event dates and times, or is that for a different ticket?

In app/.vscode/settings.json:
> @@ -0,0 +1,3 @@
+{
+    "java.configuration.updateBuildConfiguration": "interactive"
+}

As well as the .vscode files, please can you also remove the file below (app/.watchman-cookie-Marcins-MacBook-Air.local-872-124) from the pull request.

In app/src/Components/Event/EventOptions.tsx:
> @@ -27,15 +27,22 @@ const EventOptions: FC<EventOptionsProps> = ({ selected }) => {
   const handleSelectedOptionChange = (
     newSelectedOption: EventsRange | 'myEvents',
   ) => {
+    console.log(newSelectedOption);

Please can you remove this line -- console.log should normally only be used while developing/debugging, then removed when you're ready to submit a pull request.

In app/src/Components/Event/EventOptions.tsx:
>      navigate('Events', {
       screen: 'Events',
       selectedOption: newSelectedOption,
     })
   }
 
+  // const clickPastEvents = () => {

Please can you remove these unneeded commented out lines too. (When you submit a pull request, one of the steps to do is a code self-review -- it's a good idea to look through all your changes here in GitHub and double-check you haven't left in any unneeded code like this.)

In app/src/Containers/EventsContainer.tsx:
> @@ -52,18 +53,26 @@ const EventsContainer = (props: {
     fetchAllUpcomingEvents,
     { data: allUpcomingEvents },
   ] = useLazyFetchAllUpcomingEventsQuery()
+
+  const [
+    fetchAllPastEvents,
+    { data: allPastEvents},
+  ] = useLazyFetchAllPastEventsQuery() // added to useLazyFetch re PastEvents - need to investigate more

Can you remove the comment about needing to investigate more -- assume this was just for you while developing?

In app/src/Containers/EventsContainer.tsx:
>    const dispatch = useDispatch()
   const [eventsSearch, setEventsSearch] = useState<
     EventSearchInterface | undefined
   >()
   const [selectedOption, setSelectedOption] = useState<
     EventsRange | 'myEvents'
-  >(EventsRange.Upcoming)
+    >(EventsRange.Upcoming) // (EventsRange.Upcoming)

Think we don't need the comment on this line as it just repeats what's in the brackets?

In app/src/Containers/EventsContainer.tsx:
> @@ -84,12 +93,13 @@ const EventsContainer = (props: {
     setEventsSearch(props.route.params?.search)
   }, [props.route.params])
 
+  // Bellow <EventOptions selected={EventsRange.Upcoming} /> changed to selected={selectedOption}

This comment isn't needed here

In app/src/Containers/EventsContainer.tsx:
>        <SafeAreaView>
         <Text>Loading...</Text>
       </SafeAreaView>
     )
   }
 }
 
-export default EventsContainer
+export default EventsContainer;

Please can you remove the semi-colon here -- the coding style for the front-end app is no semi-colons for line endings

In app/src/Containers/EventsContainer.tsx:
> +  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])
+
+  const eventToShow = () => {
+    if(eventsSearch){ 
+      setEventsToShow(eventsSearch.results)
+    } else if (!eventsSearch && allUpcomingEvents && selectedOption === EventsRange.Upcoming){
+      setEventsToShow(allUpcomingEvents as Events)
+    } else if (!eventsSearch && allUpcomingEvents && allPastEvents && selectedOption === EventsRange.Past){
+      setEventsToShow(allPastEvents as Events)
+    }
+  } // Logic for deciding which event to render
+
+  if(eventsToShow){
+    if (allUpcomingEvents || eventsSearch) {

I don't think the logic on this line (and its closing bracket on line 159) are needed. (Did you plan to include an else for past events which you've removed?)

**Review by joefhall — CHANGES_REQUESTED** [review]
Hi Marcin, looks like it's working well showing past events without the date and time when I test out the app.

There are still some places where the code needs tidying up, or some logic correcting - please see comments.

**Comment by M4RC1N76** [comment]
Thanks for pointing out all the issues Joe. I'll check all files mentioned above on Thursday. I'll see you tomorrow at 19:30 stand up. Thanks again. See you tomorrow. 

Sent from Yahoo Mail on Android 
 
  On Tue, 4 Oct 2022 at 21:01, Joe Derry ***@***.***> wrote:   


@joefhall requested changes on this pull request.

Hi Marcin, looks like it's working well showing past events without the date and time when I test out the app.

There are still some places where the code needs tidying up, or some logic correcting - please see comments.

In app/src/Components/Event/EventReturnedList.tsx:
>  `
 const NoneFound = styled.Text`
   font-size: 18px;
   margin: 15px 15px 0px 15px;
 `
-
-const EventReturnedList: FC<EventReturnedListProps> = ({ data }) => {
+//EventsRange added to include past and upcoming Events

I don't think we need to leave in comments like this that document a change (comments are mostly useful for lasting context/explanation that's needed)

In app/src/Components/Event/EventSummary.tsx:
> @@ -12,6 +12,7 @@ import { Event } from '@/Services/modules/events'
 
 interface EventSummaryProps {
   event: Event
+  hideDateTime?: boolean // it can be a Boolean or Not

The ? means this property may exist or not (rather than being a boolean or not), so would suggest amending or removing this comment

In app/src/Components/Event/EventSummary.tsx:
> @@ -28,8 +29,8 @@ const EventInfo = styled.View`
 const RightColumn = styled.View`
   display: flex;
 `
-
-const EventSummary: FC<EventSummaryProps> = ({ event }) => (
+// full event and hideDateTime bellow

Not sure this comment is needed?

In app/src/Components/Event/EventSummary.tsx:
>      </RightColumn>
   </EventInfo>
 )
+// Above !hideDateTime with date and time to hide ('!' is a non-Null assertion)

It's up to you, but not sure whether this comment is needed?

In app/src/Containers/EventsContainer.tsx:
> @@ -85,11 +93,13 @@ const EventsContainer = (props: {
   }, [props.route.params])
 
   const EventList: FC<EventProps> = ({ data }) => {
+    console.log('selectedOption', selectedOption);

Please can you remove this console.log We shouldn't usually leave in console logging in a pull request.

In app/src/Containers/EventsContainer.tsx:
>        </SafeArea>
     )
   }
 
-  if (allUpcomingEvents || eventsSearch) {
-    const eventsToShow = eventsSearch
-      ? eventsSearch.results
-      : (allUpcomingEvents as Events)
+  const [eventsToShow, setEventsToShow] = useState<Events>() // holding state for which event are being rendered
+
+  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])
+
+  const eventToShow = () => {
+    if(eventsSearch){ 
+      setEventsToShow(eventsSearch.results)
+    } else if (!eventsSearch && allUpcomingEvents && selectedOption === EventsRange.Upcoming){

As mentioned in previous review, this line doesn't need the !eventsSearch && part of this condition -- in line 144 you're already checking eventsSearch and we're now in an else if alternative to it

In app/src/Containers/EventsContainer.tsx:
> -  if (allUpcomingEvents || eventsSearch) {
-    const eventsToShow = eventsSearch
-      ? eventsSearch.results
-      : (allUpcomingEvents as Events)
+  const [eventsToShow, setEventsToShow] = useState<Events>() // holding state for which event are being rendered
+
+  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])
+
+  const eventToShow = () => {
+    if(eventsSearch){ 
+      setEventsToShow(eventsSearch.results)
+    } else if (!eventsSearch && allUpcomingEvents && selectedOption === EventsRange.Upcoming){
+      setEventsToShow(allUpcomingEvents as Events)
+    } else if (!eventsSearch && allUpcomingEvents && allPastEvents && selectedOption === EventsRange.Past){

This line also doesn't need the !eventsSearch && part of this condition.

And it shouldn't include && allUpcomingEvents on this line -- we should be able to show allPastEvents even if, for instance, upcoming events haven't loaded successfully.

In app/src/Containers/EventsContainer.tsx:
>        </SafeArea>
     )
   }
 
-  if (allUpcomingEvents || eventsSearch) {
-    const eventsToShow = eventsSearch
-      ? eventsSearch.results
-      : (allUpcomingEvents as Events)
+  const [eventsToShow, setEventsToShow] = useState<Events>() // holding state for which event are being rendered
+
+  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])
+
+  const eventToShow = () => {
+    if(eventsSearch){ 
+      setEventsToShow(eventsSearch.results)
+    } else if (!eventsSearch && allUpcomingEvents && selectedOption === EventsRange.Upcoming){
+      setEventsToShow(allUpcomingEvents as Events)

as Events isn't needed here as Typescript already knows the type of allUpcomingEvents

In app/src/Containers/EventsContainer.tsx:
> -    const eventsToShow = eventsSearch
-      ? eventsSearch.results
-      : (allUpcomingEvents as Events)
+  const [eventsToShow, setEventsToShow] = useState<Events>() // holding state for which event are being rendered
+
+  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])
+
+  const eventToShow = () => {
+    if(eventsSearch){ 
+      setEventsToShow(eventsSearch.results)
+    } else if (!eventsSearch && allUpcomingEvents && selectedOption === EventsRange.Upcoming){
+      setEventsToShow(allUpcomingEvents as Events)
+    } else if (!eventsSearch && allUpcomingEvents && allPastEvents && selectedOption === EventsRange.Past){
+      setEventsToShow(allPastEvents as Events)

as Events isn't needed here as Typescript already knows the type of allPastEvents

In app/src/Containers/EventsContainer.tsx:
>        </SafeArea>
     )
   }
 
-  if (allUpcomingEvents || eventsSearch) {
-    const eventsToShow = eventsSearch
-      ? eventsSearch.results
-      : (allUpcomingEvents as Events)
+  const [eventsToShow, setEventsToShow] = useState<Events>() // holding state for which event are being rendered
+
+  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])

allPastEvents and eventsSearch also need to be added into the array here, as they will affect the result of eventToShow

I don't think props.route.params should be inside square brackets?

File: app/src/Containers/EventsContainer.tsx
Comment (M4RC1N76) [reply app/src/Containers/EventsContainer.tsx:141]: I tried removing the square brackets, however the app showed the wrong results i.e. when I clicked on **Past** it showed me the **Upcoming** events and vice verse. I have left the code unchanged with the square brackets around [props.route.params].

**Review by DougieWougie — APPROVED** [review]

**Comment by M4RC1N76** [comment]
The fragments, allow the use of multiple child elements. I got carried away and was replying in the slack channel.
    On Friday, 7 October 2022 at 13:33:24 BST, Dougie Richardson ***@***.***> wrote:  
 
 


@DougieWougie approved this pull request.

In app/src/Components/Event/EventSummary.tsx:
> @@ -42,10 +43,18 @@ const EventSummary: FC<EventSummaryProps> = ({ event }) => (
     />
     <RightColumn>
       <Title text={event.name} type="list" />
-      <EventDate date={event.date} />
-      <EventTime time={event.time} />
+      {
+        !hideDateTime && (
+          <>

What do these angle brackets do?

File: app/src/Components/Event/EventSummary.tsx
Comment (M4RC1N76) [reply app/src/Components/Event/EventSummary.tsx:48]: Using these brackets we can use multiple child elements.

File: app/src/Containers/EventsContainer.tsx
Comment (joefhall) [inline app/src/Containers/EventsContainer.tsx:141]: Hi Marcin, I've taken another look at this line as I didn't feel it was right.  The key thing here is to think about how `useEffect` works -- it runs its code in its first parameter (i.e. `eventToShow()`) any time one of the things in the array in its second parameter (line 141) changes.

If we look at the `eventToShow` function (lines 143-151), there are several variables that could affect the outcome of the logic of the `if` / `else if` statements here -- `eventsSearch`, `allUpcomingEvents`, `allPastEvents` and `selectedOption`.

Therefore, logically, we need to include all of these variables in the array that's the second parameter of `useEffect` on line 141 (in order to make `eventToShow()` run again) -- so this array should be `[eventsSearch, allUpcomingEvents, allPastEvents, selectedOption]`  I've also tried this out locally to check it works.

Hope that makes sense?

(I'm not sure quite what the effect was of putting `props.route.params` inside its own array -- `[props.route.params]` -- but it's definitely not the right answer here, even if it seemed to work in practice.  Always good to go back to first principles when something's unclear, to make sure we know why something should work.)

**Review by joefhall — CHANGES_REQUESTED** [review]

**Comment by M4RC1N76** [comment]
Thank you for the insight Joe. Dougie was also asking about the possibility to test the app before merge with main branch. I'll investigate that, however I am so busy this weekend at work that the soonest I can check it is Monday. Thanks again for the informations. 
Regards Marcin 
Sent from Yahoo Mail on Android 
 
  On Sat, 8 Oct 2022 at 15:40, Joe Derry ***@***.***> wrote:   


@joefhall requested changes on this pull request.

In app/src/Containers/EventsContainer.tsx:
>        </SafeArea>
     )
   }
 
-  if (allUpcomingEvents || eventsSearch) {
-    const eventsToShow = eventsSearch
-      ? eventsSearch.results
-      : (allUpcomingEvents as Events)
+  const [eventsToShow, setEventsToShow] = useState<Events>() 
+
+  useEffect (() => {
+    eventToShow()
+  }, [allUpcomingEvents, [props.route.params]])

Hi Marcin, I've taken another look at this line as I didn't feel it was right. The key thing here is to think about how useEffect works -- it runs its code in its first parameter (i.e. eventToShow()) any time one of the things in the array in its second parameter (line 141) changes.

If we look at the eventToShow function (lines 143-151), there are several variables that could affect the outcome of the logic of the if / else if statements here -- eventsSearch, allUpcomingEvents, allPastEvents and selectedOption.

Therefore, logically, we need to include all of these variables in the array that's the second parameter of useEffect on line 141 (in order to make eventToShow() run again) -- so this array should be [eventsSearch, allUpcomingEvents, allPastEvents, selectedOption] I've also tried this out locally to check it works.

Hope that makes sense?

(I'm not sure quite what the effect was of putting props.route.params inside its own array -- [props.route.params] -- but it's definitely not the right answer here, even if it seemed to work in practice. Always good to go back to first principles when something's unclear, to make sure we know why something should work.)

File: app/src/Containers/EventsContainer.tsx
Comment (M4RC1N76) [reply app/src/Containers/EventsContainer.tsx:141]: Thanks for the insight Joe. I have replied to this message a day ago from my mobile when at work, however it did not show up here (I am guessing you have received an email with the reply which was not registered here). Thanks again for the detailed explanation.

**Comment by joefhall** [comment]
No worries @M4RC1N76 -- hopefully you're ok to make that last change?  

Dougie's comment about testing (answering a question of mine) was on Rosie's other PR (PR 92 for SVA 213), not one you need to worry about.

**Comment by M4RC1N76** [comment]
Haha, yes I have noticed Dougie's comment was associated with Rosie's PR. I need to pay more attention to the details when replying from my android phone (here it all looks like a chain of comments to me). I am out right now,  however I will do my best to commit all the changes suggested by you as soon as so you can check if everything is fine and finaly do the marges. Thanks again for your insightful comments Joe.

Sent from Yahoo Mail on Android 
 
  On Tue, 11 Oct 2022 at 8:35, Joe Derry ***@***.***> wrote:   


No worries @M4RC1N76 -- hopefully you're ok to make that last change?

Dougie's comment about testing (answering a question of mine) was on Rosie's other PR (PR 92 for SVA 213), not one you need to worry about.

**Comment by joefhall** [comment]
No worries, yep it can get a little messy using email to respond to GitHub comments -- e.g. it can copy in quoted replies, not just your new message.  I find it helpful to see notifications in email but respond in GitHub -- totally up to you though.

**Comment by DougieWougie** [comment]
GitHub's app is handy too, I prefer to respond via that @joefhall is right email is noisy.

**Comment by M4RC1N76** [comment]
I often try to reply as fast as I can hence use of android phone. I remember Dougie recommended GitHub so I'll follow that path in the future. 

Sent from Yahoo Mail on Android 
 
  On Tue, 11 Oct 2022 at 13:45, Dougie ***@***.***> wrote:   


GitHub's app is handy too, I prefer to respond via that @joefhall is right email is noisy.

**Review by joefhall — APPROVED** [review]
Well done!

**Comment by M4RC1N76** [comment]
Thank you @joefhall for your assistance, see you tomorrow at the stand up .

**Pull request merged**
